File: app/src/debug/java/com/sakuna63/tumblrin/application/di/module/DebugApiModule.java
package com.sakuna63.tumblrin.application.di.module;

import com.facebook.stetho.okhttp3.StethoInterceptor;
import com.sakuna63.tumblrin.application.di.qualifier.NetworkInterceptor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.multibindings.IntoSet;
import okhttp3.Interceptor;
import okhttp3.logging.HttpLoggingInterceptor;

@Module
public class DebugApiModule extends ApiModule {

    @Provides
    @Singleton
    @NetworkInterceptor
    @IntoSet
    Interceptor httpLoggingInterceptor() {
        return new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY);
    }

    @Provides
    @Singleton
    @NetworkInterceptor
    @IntoSet
    Interceptor stethoInterceptor() {
        return new StethoInterceptor();
    }

}

File: app/src/debug/java/com/sakuna63/tumbin/application/di/module/DebugApiModule.kt
package com.sakuna63.tumbin.application.di.module

import com.facebook.stetho.okhttp3.StethoInterceptor
import oauth.signpost.OAuthConsumer
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

class DebugApiModule : ApiModule() {

    override fun okHttpClient(consumer: OAuthConsumer): OkHttpClient {
        val builder = super.okHttpClient(consumer).newBuilder()
        return addDebugInterceptors(builder).build()
    }

    override fun unsafeOkHttpClient(): OkHttpClient {
        val builder = super.unsafeOkHttpClient().newBuilder()
        return addDebugInterceptors(builder).build()
    }

    private fun addDebugInterceptors(builder: OkHttpClient.Builder): OkHttpClient.Builder {
        return builder
                .addNetworkInterceptor(StethoInterceptor())
                .addNetworkInterceptor(buildHttpLoggingInterceptor())
    }

    private fun buildHttpLoggingInterceptor() =
            HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)
}
